lif/lif.go

```diff
@@ -90,4 +130,205 @@ func LifBuild() {
+		}
+	}
+
+	return yaml.Marshal(compose)
+}
+
+// generate a docker compose builder for the postgres resource
+func GeneratePostgresComposeMap(dbname string, resourceFields map[string]any) (map[string]any, error) {
+	user := generateRandomString(16)
+	password := generateRandomString(16)
+
+	println("dbname", dbname)
+	println("user", user)
+	println("password", password)
+
+	// Store concrete values in composeResources
+	composeResources[dbname] = map[string]any{
+		"url":      "postgres://" + user + ":" + password + "@mydb:5432/" + dbname,
+		"user":     user,
+		"password": password,
+	}
+
+	compose := map[string]any{
+		"image": "postgres:latest",
+		"environment": map[string]string{
+			"POSTGRES_USER":     user,
+			"POSTGRES_PASSWORD": password,
+			"POSTGRES_DB":       dbname,
+		},
+		"ports": []string{"5432:5432"},
+		"healthcheck": map[string]any{
+			"test":     []string{"CMD-SHELL", fmt.Sprintf("psql -U %s -d %s -c 'SELECT 1;'", user, dbname)},
+			"interval": "5s",
+			"timeout":  "5s",
+			"retries":  5,
+		},
+	}
+
+	return compose, nil
+}
+
+func GenerateEC2ComposeMap(resourceMap map[string]any) (map[string]any, error) {
+	envVars := resourceMap["env_vars"].(map[string]string)
+	resolvedEnvVars := make(map[string]string)
+	dependencies := make([]string, 0)
+	dockerfilePath := resourceMap["dockerfile"].(string)
+	buildContext := filepath.Dir(dockerfilePath)
+	dockerfileName := filepath.Base(dockerfilePath)
+
+	// resolve references and collect dependencies
+	for k, v := range envVars {
+		ref := v
+		resolvedEnvVars[k] = resolveLocalRef(ref)
+
+		// if this env var references another resource, it's a dependency
+		parts := strings.Split(ref, ":")
+		if len(parts) == 3 && parts[0] == "ref" {
+			dependencies = append(dependencies, parts[1])
+		}
+	}
+
+	compose := map[string]any{
+		"environment": resolvedEnvVars,
+		"ports":       resourceMap["ports"],
+		"depends_on":  makeHealthcheckDependencies(dependencies),
+		"build": map[string]string{
+			"context":    buildContext,
+			"dockerfile": dockerfileName,
+		},
+	}
+
+	return compose, nil
+}
+
+func resolveLocalRef(ref string) string {
+	// ref format is "ref:resourceName:property"
+	parts := strings.Split(ref, ":")
+	if len(parts) != 3 || parts[0] != "ref" {
+		return ref
+	}
+
+	resourceName := parts[1]
+	property := parts[2]
+
+	// Check composeResources first
+	if composeRes, exists := composeResources[resourceName]; exists {
+		if val, ok := composeRes[property]; ok {
+			return val.(string)
+		}
+	}
+
+	// Then check regular resources
+	if res, exists := resources[resourceName]; exists {
+		if resourceSpecs, ok := res.(map[string]any); ok && resourceSpecs["type"] == "postgres" {
+			exposes := resourceSpecs["exposes"].(NeonPostgresExposes)
+
+			switch property {
+			case "url":
+				return exposes.Url
+			case "user":
+				return exposes.User
+			case "password":
+				return exposes.Password
+			}
+		}
+	}
+
+	// if we can't resolve it, return the original ref??
```

Comment:
This doesn't sound right. We should error. For stuff that isn't a ref we should have other prefixes.